src/driver/postgres/PostgresQueryRunner.ts

```diff
@@ -1542,7 +1541,23 @@ export class PostgresQueryRunner
                     ),
                 )
             }
-
+            // Handle length changes without recreating the column (e.g., varchar(n) -> varchar(m))
+            if (newColumn.length !== oldColumn.length) {
```

Comment:
The block above this does exactly the same thing for precision and scale. It would be better to combine these cases.